### ISSUE TITLE
[Bug] Crafting module doesn't seem to work when there's multiple recipes for the block.

### ISSUE THREAD

**Issue by SubjectHawk66** [Tschipp/BuildersBag#9]
**Expected Behavior**
The steel blocks in the bag are crafted into scaffold
**Actual Behavior**
The bag seems to think, well I don't the slabs therefor I cannot craft it (recipes in screenshot). How I'm assuming the bag work is it uses the first recipe to craft it, but it doesn't seem to like it when there's multiple recipes. It also doesn't work with stone bricks.

![2020-05-11_17 44 13](https://user-images.githubusercontent.com/35781469/81615572-cf9db300-93af-11ea-9d84-229148cc9e09.png)
![2020-05-11_17 45 25](https://user-images.githubusercontent.com/35781469/81615574-d0364980-93af-11ea-972b-4aaa668ef7e1.png)
![2020-05-11_17 49 27](https://user-images.githubusercontent.com/35781469/81615575-d0364980-93af-11ea-9942-e5aeaff44b58.png)
![2020-05-11_17 49 30](https://user-images.githubusercontent.com/35781469/81615576-d0364980-93af-11ea-90e0-f061159099a8.png)

Basically it didn't work wither either when I had it in the bag and placed it with the modules enabled.

**Steps to Reproduce**
Put steel blocks (immersive engineering) and a steel slab in ATM3: Remix or try with other blocks like stone bricks.
**1.12.2, Builder's Bag 1.1.0.7 (not latest but I don't think it would've been fixed), Singleplayer

**Screenshots provided**

**Comment by SubjectHawk66** [comment]
Unless I'm having a weird bug where the module isn't working, I just wanna make sure.

**Comment by SubjectHawk66** [comment]
Yah the module just isn't working so idk what's happening I'll try updating the mod.

**Comment by SubjectHawk66** [comment]
#7 Seems to be my issue but it's not working ever for me, but everything I am trying has multiple recipes so. It worked for a little bit though because I had some loose items that were ingredients to scaffolding. EDIT: It worked for something with a single recipe so I think I found the issue.

**Comment by Tschipp** [comment]
Yes that is interesting. It's (obviously) working fine in my lightly modded testworld, but I guess I'll try a heavier one to figure this out.

**Comment by Tschipp** [comment]
Hmm, just tried the recipe you mentioned and it works for me. Guess I'll try downloading your pack and see what's up there.

**Comment by Tschipp** [comment]
Okay I think I might see the issue, I also got it when I used slabs for the materials (it worked with blocks). I'll poke around in the crafting module, but I'm scared to go back in there because it's so complex lol

**Comment by Tschipp** [comment]
Yeah, probably gonna need a rewrite of how crafting is done.

**Comment by SubjectHawk66** [comment]
Yah it did work sometimes but it seemed to only target the first recipe for the block instead of the most logical or at least the one it had the materials for. I’m no coder but this seems like a very complex thing to code. It would have to try to use all of the recipes to craft the block or at least try the one with the materials in the bag. Or something.

**Comment by Tschipp** [comment]
Yeah it's very complex, I use a graph structure to build the recipes but I may need to rethink it.

**Comment by Tschipp** [comment]
Okay, I think I fixed this one. I had to do some rewriting of core systems, but it works now (and has a better performance than before too, yay!)

**Comment by SubjectHawk66** [comment]
Glad I could add to the issue.

**Comment by Tschipp** [comment]
Keep in mind though that it is very possible that the bag still creates some scaffolding slabs instead of just using the steel and rods, because I randomize what recipe gets picked to make a stack (for obvious reasons)
Edit: Actually, no: it is not necessary that I randomize, I can just switch to a different recipe lol. Implemented that now, so you should always see consistent behavior now.

**Comment by SubjectHawk66** [comment]
I was about to suggest that it just switch. Since I assumed that it would be possible.

**Comment by Tschipp** [comment]
Please let me know if the most recent update fixed this for you.

**Comment by SubjectHawk66** [comment]
Alrighty I’ll try it today.

**Comment by SubjectHawk66** [comment]
Remember when I said I'd try it 9 days ago? Well I forgot, just got around to testing the feature. Now it's not working for the first listed recipe either for me. I've looked at other issues and this could be just because of my modpack. Also the chisel module isn't working either.

**Comment by Tschipp** [comment]
Chisel module isn't working? Do you have a chisel (with durability) in the internal inventory of the chisel module?

**Comment by SubjectHawk66** [comment]
Yah I made sure.

**Comment by Tschipp** [comment]
Interesting. Could you show a gif or video of what you're trying to do?

**Comment by SubjectHawk66** [comment]
https://drive.google.com/open?id=14NoLaxYJfBh0QMUXRqxobrWhVRJ5_oEa as you can see, neither are working when I test on even simple materials with one step.

**Comment by Tschipp** [comment]
Huh, that is very strange. Just to be sure, is there actually no block created? Could you try by selecting the block and placing it with the bag?

**Comment by SubjectHawk66** [comment]
Uh wtf, it just started working. I just get massive lag spikes when it crafts stuff now.

Edit: Here's my F3 when I get the spikes if it helps. https://drive.google.com/open?id=1EIoBn8b0TnhAtafnz_0GIPn6KDBBwQ4C

**Comment by Tschipp** [comment]
I think I probably know why, it's related to a module priority that I implemented recently, but I did it wrongly so that it did the exact opposite of what I wanted. I have fixed it in dev, but it's not out yet.

**Comment by SubjectHawk66** [comment]
Alright cool. I can live without it for a bit.
